kernel/init.c

```diff
@@ -279,11 +278,22 @@ void __weak main(void)
 /* LCOV_EXCL_STOP */
 
 #if defined(CONFIG_MULTITHREADING)
-static void init_idle_thread(struct k_thread *thread, k_thread_stack_t *stack)
+static void init_idle_thread(int i)
 {
+	struct k_thread *thread = &z_idle_threads[i];
+	k_thread_stack_t *stack = z_idle_stacks[i];
+
+#ifdef CONFIG_THREAD_NAME
+	char tname[8];
+
+	snprintk(tname, 8, "idle %02d", i);
```

Comment:
`snprintk(tname, sizeof(tname), "idle %02d", i);`

Reply:
Do we really need fixed 2-digit numbers? (e.g. `idle 00`, `idle 01`, ...) What if we have more than 99 cores? Wouldn't it be better to just name them `idle 0`, `idle 1`, ...?

Reply:
> What if we have more than 99 cores

On a microcontroller OS? Really?

Reply:
> On a microcontroller OS? Really?

Probably not, though with neural network being a thing nowadays, that probably isn't all that far-fetched (I mean we already have 64; I suspect 128 or even 1024 is not that far away).